proxmoxtf/resource_virtual_environment_network.go

```diff
@@ -0,0 +1,373 @@
+	config := m.(providerConfiguration)
+	veClient, err := config.GetVEClient()
+
+	if err != nil {
+		return err
+	}
+	nodeName := d.Get(mkResourceVirtualEnvironmentNetworkNodeName).(string)
+	iface := d.Get(mkResourceVirtualEnvironmentNetworkIface).(string)
+
+	d.SetId(fmt.Sprintf("%s_network_%s", nodeName, strings.ReplaceAll(iface, ".", "_")))
+
+	err = veClient.DeleteNetworkInterface(nodeName, iface)
+
+	if err != nil {
+		return err
+	}
+
+	d.SetId("")
+
+	return nil
+}
+
+func resourceVirtualEnvironmentNetworkUpdate(d *schema.ResourceData, m interface{}) error {
+	config := m.(providerConfiguration)
+	veClient, err := config.GetVEClient()
+
+	if err != nil {
+		return err
+	}
+
+	nodeName := d.Get(mkResourceVirtualEnvironmentNetworkNodeName).(string)
+	iface := d.Get(mkResourceVirtualEnvironmentNetworkIface).(string)
+
+	d.SetId(fmt.Sprintf("%s_network_%s", nodeName, strings.ReplaceAll(iface, ".", "_")))
+
+	body := getBody(d, true)
+	err = veClient.UpdateNetworkInterface(nodeName, iface, body)
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func resourceVirtualEnvironmentNetworkCreate(d *schema.ResourceData, m interface{}) error {
+	config := m.(providerConfiguration)
+	veClient, err := config.GetVEClient()
+
+	if err != nil {
+		return err
+	}
+	nodeName := d.Get(mkResourceVirtualEnvironmentNetworkNodeName).(string)
+	iface := d.Get(mkResourceVirtualEnvironmentNetworkIface).(string)
+
+	d.SetId(fmt.Sprintf("%s_network_%s", nodeName, strings.ReplaceAll(iface, ".", "_")))
+
+	body := getBody(d, false)
+	err = veClient.CreateNetworkInterface(nodeName, body)
+
+	if err != nil {
+		return err
+	}
+
+	return resourceVirtualEnvironmentNetworkRead(d, m)
+}
+
+func resourceVirtualEnvironmentNetworkRead(d *schema.ResourceData, m interface{}) error {
+	config := m.(providerConfiguration)
+	veClient, err := config.GetVEClient()
+
+	if err != nil {
+		return err
+	}
+
+	nodeName := d.Get(mkResourceVirtualEnvironmentNetworkNodeName).(string)
+	iface := d.Get(mkResourceVirtualEnvironmentNetworkIface).(string)
+
+	network, err := veClient.GetNetworkInterface(nodeName, iface)
+
+	if err != nil {
+		return err
+	}
+
+	d.SetId(fmt.Sprintf("%s_network_%s", nodeName, strings.ReplaceAll(iface, ".", "_")))
+
+	d.Set(mkResourceVirtualEnvironmentNetworkType, network.Type)
+	d.Set(mkResourceVirtualEnvironmentContainerNetworkInterface, network.Iface)
+	d.Set(mkResourceVirtualEnvironmentNetworkAddress, network.Address)
+	d.Set(mkResourceVirtualEnvironmentNetworkCIDR, network.CIDR)
+	d.Set(mkResourceVirtualEnvironmentNetworkNetmask, network.Netmask)
+	d.Set(mkResourceVirtualEnvironmentNetworkAutostart, network.Autostart)
+	d.Set(mkResourceVirtualEnvironmentNetworkBridgePorts, network.BridgePorts)
+	d.Set(mkResourceVirtualEnvironmentNetworkGateway, network.Gateway)
+
+	return nil
+}
+
+func getBody(d *schema.ResourceData, isUpdate bool) *proxmox.VirtualEnvironmentNetworkInterfaceCreateRequestBody {
```

Comment:
Consider renaming to `resourceVirtualEnvironmentNetworkGetBody` to prevent name collision across resources.